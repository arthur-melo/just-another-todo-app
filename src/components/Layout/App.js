import React from 'react';

import './App.css';

import Header from '../Header/Header';
import ContentContainer from '../../containers/ContentContainer';
import Footer from '../Footer/Footer';

const App = () => (
  <div className="app__container">
    <Header className="app__header" />
    <ContentContainer className="app__content" />
    <Footer className="app__footer" />
  </div>
);

export default App;
