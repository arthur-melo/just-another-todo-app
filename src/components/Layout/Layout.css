.layout__header {
  min-height: 10vh;
}

.layout__footer {
  min-height: 10vh;
}
