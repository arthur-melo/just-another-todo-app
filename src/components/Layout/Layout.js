import PropTypes from 'prop-types';

import './Layout.css';

import Header from './Header/Header';
import Footer from './Footer/Footer';

const layoutPropTypes = {
  children: PropTypes.node.isRequired,
};

const Layout = props => {
  const { children } = props;

  return (
    <div className="d-flex flex-column w-100 h-100">
      <header className="layout__header flex-shrink-0 w-100">
        <Header />
      </header>
      <main className="flex-grow-1 flex-shrink-0 mt-1">{children}</main>
      <footer className="layout__footer flex-shrink-0 w-100">
        <Footer />
      </footer>
    </div>
  );
};

Layout.propTypes = layoutPropTypes;

export default Layout;
