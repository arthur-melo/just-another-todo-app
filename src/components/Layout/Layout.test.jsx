import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

import Layout from './Layout';

describe('Layout', () => {
  it('should render the Layout component', () => {
    const child = <div>Children</div>;
    const component = <Layout>{child}</Layout>;

    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });
});
