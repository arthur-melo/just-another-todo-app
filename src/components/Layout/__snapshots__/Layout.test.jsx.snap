// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > should render the Layout component 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column w-100 h-100"
  >
    <header
      class="layout__header flex-shrink-0 w-100"
    >
      <div
        class="d-flex justify-content-center align-items-center w-100 h-100 bg-light"
      >
        <nav
          class="navbar"
        >
          <span
            class="navbar-brand m-0"
          >
            <svg
              alt="List emoji"
              aria-hidden="true"
              class="svg-inline--fa fa-rectangle-list fa-lg d-inline-block"
              color=""
              data-icon="rectangle-list"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 96C0 60.7 28.7 32 64 32l448 0c35.3 0 64 28.7 64 64l0 320c0 35.3-28.7 64-64 64L64 480c-35.3 0-64-28.7-64-64L0 96zM128 288a32 32 0 1 0 0-64 32 32 0 1 0 0 64zm32-128a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zM128 384a32 32 0 1 0 0-64 32 32 0 1 0 0 64zm96-248c-13.3 0-24 10.7-24 24s10.7 24 24 24l224 0c13.3 0 24-10.7 24-24s-10.7-24-24-24l-224 0zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24l224 0c13.3 0 24-10.7 24-24s-10.7-24-24-24l-224 0zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24l224 0c13.3 0 24-10.7 24-24s-10.7-24-24-24l-224 0z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span
            class="header__text align-top fs-5"
          >
            Just Another Todo App
          </span>
        </nav>
      </div>
    </header>
    <main
      class="flex-grow-1 flex-shrink-0 mt-1"
    >
      <div>
        Children
      </div>
    </main>
    <footer
      class="layout__footer flex-shrink-0 w-100"
    >
      <div
        class="bg-light text-center d-flex h-100 justify-content-center align-items-center p-1"
      >
        <div
          class="wrapper"
        >
           
          <p
            class="w-100 m-0"
          >
            This app runs offline and doesn't use analytics.
          </p>
          <p
            class="w-100 m-0"
          >
            Your data is safe in your device!
          </p>
          <p
            class="w-100 m-0"
          >
            Made with ❤️ by 
            <a
              class="text-info"
              href="https://github.com/arthur-melo"
              rel="noopener noreferrer"
              target="_blank"
            >
              @arthur-melo
            </a>
          </p>
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
